scipy/optimize/_brent.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import division
+
+from math import sqrt
+
+from numpy import finfo
+
+_sqrt_epsilon = sqrt(finfo(float).eps)
+# about 0.3819660112501051
+_golden = 0.5 * (3.0 - sqrt(5.0))
+
+
+def find_minimum(f, a, b, rtol=_sqrt_epsilon, atol=_sqrt_epsilon, maxiter=500):
+    """Seeks a local minimum of a function f in a closed interval [a, b] via
+    Brent's method.
+
+    Given a function f with a minimum in the interval the a <= b,
+    seeks a local minima using a combination of golden section search and
+    successive parabolic interpolation.
+
+    Let ```tol = rtol * abs(x0) + atol```, where ```x0``` is the best guess
+    found so far. It converges if evaluating a next guess would imply
+    evaluating ```f``` at a point that is closer than ```tol``` to a previously
+    evaluated one or if the number of iterations reaches ```maxiter```.
+
+    Args:
+        f (object): Objective function to be minimized.
+        a, b (float): endpoints of the interval a <= b.
+        rtol (float): relative tolerance. Defaults to 1.4901161193847656e-08.
+        atol (float): absolute tolerance. Defaults to 1.4901161193847656e-08.
+        maxiter (int): maximum number of iterations
+
+
+    Returns:
+        float: best guess for the minimum of f.
+        float: value of f evaluated at the best guess.
+        int: number of iterations performed.
+
+    References:
+        - http://people.sc.fsu.edu/~jburkardt/c_src/brent/brent.c
```

Comment:
This code is licensed under the LGPL--i.e. it's incompatible with Scipy. If your code is based on it that's a problem. Numerical Recipies is also questionable...